runtime/port/sysvipc/j9sharedhelper.c

```diff
@@ -217,33 +217,61 @@ cleanSharedMemorySegments(struct J9PortLibrary* portLibrary)
 #define OUTPUTBUFSIZE 256
 
 	OMRPortLibrary *omrPortLibrary = OMRPORT_FROM_J9PORT(portLibrary);
-	char *ipcsField[IPCS_NO_FIELDS];
-	char outputBuf[OUTPUTBUFSIZE];
-	char *bufferPtr;
-	char *bufferEndPtr;
-	FILE *ipcsOutput;
+	char *ipcsField[IPCS_NO_FIELDS] = {0};
+	char outputBuf[OUTPUTBUFSIZE] = {0};
+	char *bufferPtr = NULL;
+	char *bufferEndPtr = NULL;
+	FILE *ipcsOutput = NULL;
 	int noHeaderLines = IPCS_NO_HEADER_LINES;
 	char processOwner[OUTPUTBUFSIZE] = "";
-	int i;
+	int i = 0;
+	IDATA result = 0;
+
+	OMRPORT_ACCESS_FROM_J9PORT(portLibrary);
 
 #if defined(J9ZOS390)
 	pid_t pid = -1;
-	int childStatus;
+	int childStatus = 0;
 #endif
 
 	Trc_PRT_shared_cleanSharedMemorySegments_entry();
-
-	omrPortLibrary->sysinfo_get_username(omrPortLibrary, processOwner, OUTPUTBUFSIZE);
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+	/* This is equivalent to !isCheckpointAllowed().
+	 * https://github.com/eclipse-openj9/openj9/issues/15800
+	 */
+	if (portLibrary->nanoTimeMonotonicClockDelta > 0)
```

Comment:
`(portLibrary->nanoTimeMonotonicClockDelta > 0)` just means we've done a restore, it doesnt mean that we will never checkpoint again

Reply:
I probably should also state that this is equivalent to `!isCheckpointAllowed()` in `NonPortableRestoreMode` (i.e., `-XX:+CRIURestoreNonPortableMode` is specified) which is current CRIU default mode.
`nanoTimeMonotonicClockDelta` only supports one checkpoint https://github.com/eclipse-openj9/openj9/blob/ec94f811b925560131bb12c6e06374ef6ecc83d1/runtime/oti/j9port_generated.h#L391-L395
Also just realized that `nanoTimeMonotonicClockDelta` could be negative, and possible `0` though the possibility is low.
I will have to add another field within `portLibrary` indicating whether a checkpoint is allowed.

Reply:
Added `J9PortLibrary.isSafeSSSDMethod` which is equivalent to `!isCheckpointAllowed()`.
@tajila this is ready for another look.